homework/index.js

```diff
@@ -1,47 +1,238 @@
+    const list = document.querySelector('.contributors__list');
+    try {
+      const contributors = await Helper.fetchJSON(selectedRepository.contributors_url);
+      if (!contributors.length) {
+        Helper.createAndAppend('li', list, {
+          class: 'alert-warning',
+          text: 'There is no contributor for this repository!',
+        });
+        return;
+      }
+
+      contributors.forEach(contributor => {
+        const listItem = Helper.createAndAppend('li', list, {
+          class: 'contributors__item',
+        });
+        const listLink = Helper.createAndAppend('a', listItem, {
+          href: contributor.html_url,
+          target: '_blank',
+          class: 'contributors__link',
+        });
+        Helper.createAndAppend('img', listLink, {
+          class: 'contributors__avatar',
+          src: contributor.avatar_url,
+          alt: contributor.login,
+        });
+        Helper.createAndAppend('span', listLink, {
+          class: 'contributors__name',
+          text: contributor.login,
+        });
+        Helper.createAndAppend('span', listLink, {
+          class: 'contributors__badge',
+          text: contributor.contributions,
+        });
+      });
+    } catch (error) {
+      Helper.createAndAppend('li', list, { text: error.message, class: 'alert-error' });
+    }
+  }
+
+  class App {
+    constructor(select) {
+      this.select = select;
+    }
+
+    static init(data, root) {
+      createStaticComponentsFromModel(componentModel, root);
+
+      const repositories = data.sort((a, b) =>
+        a.name.toLowerCase().localeCompare(b.name.toLowerCase()),
+      );
+
+      this.select = document.querySelector('.header__select');
+
+      repositories.forEach(repository => {
+        Helper.createAndAppend('option', this.select, { text: repository.name });
+      });
+
+      this.select.addEventListener('change', () => {
+        document.querySelector('tbody').innerHTML = '';
+        document.querySelector('.contributors__list').innerHTML = '';
+        fillRepositoryDetailsTable(repositories[this.select.selectedIndex]);
+        fillContributorsList(repositories[this.select.selectedIndex]);
+      });
+
+      fillRepositoryDetailsTable(repositories[this.select.selectedIndex]);
+      fillContributorsList(repositories[this.select.selectedIndex]);
+    }
+
+    static async main(url) {
       const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+
+      try {
+        const repositories = await Helper.fetchJSON(url);
+        this.init(repositories, root);
+      } catch (err) {
+        Helper.createAndAppend('div', root, { class: 'alert-error', text: err.message });
       }
-    });
+    }
   }
 
   const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
```

Comment:
Your app doesn't handle HTTP errors correctly. For instance, try and change `orgs` in `orgsx` in this url to force an HTTP 404 error and reload the browser.